src/js/utils/events.js

```diff
@@ -351,8 +351,12 @@ export function off(elem, type, fn) {
 
   // Are we removing all bound events?
   if (!type) {
-    for (const t in data.handlers) {
-      removeType(t);
+    if (!!data && !!data.handlers) {
+      for (const t in data.handlers) {
+        if (!!data.handlers && !!data.handlers.hasOwnProperty && data.handlers.hasOwnProperty(t)) {
```

Comment:
does this mean that if `hasOwnProperty` is missing for whatever reason, we'll end up not actually removing the handler?

Reply:
I just went in and reconstructed the situation.
I called ```player.off("")``` to get into the ```if (!type)``` statement.
It had ```data.handlers``` filled up and in the loop removed one child after another. It seems like after the last own property is removed data.handler gets removed, as when it starts processing the prototype properties ```data``` actually is ```{}```. 

So in the end 
```JS
if (!type) {
  for (const t in data.handlers) {
    if (data.handlers) {
      removeType(t);
    }
  }
  return;
}
```
would totally do the trick.

Reply:
Is `player.off('')` what causing issues here? Are there other examples?

Reply:
I hit this problem at work in a more complex code base. I cannot exactly find the instance/code point that caused it.

```player.off('')``` was the most condensed trigger I could find to fit that code. It appears any falsy input would trigger the loop.